prow/statusreconciler/controller.go

```diff
@@ -268,7 +268,7 @@ func addedBlockingPresubmits(old, new map[string][]config.Presubmit) map[string]
 	for repo, oldPresubmits := range old {
 		added[repo] = []config.Presubmit{}
 		for _, newPresubmit := range new[repo] {
-			if !newPresubmit.ContextRequired() {
+			if !(newPresubmit.ContextRequired() && newPresubmit.AlwaysRun) {
```

Comment:
How are run_if_changed jobs handled? I'd expect if a new job is added with a run_if_changed that matched, the status should be set to Pending.

Reply:
yeah, I just realized that as well

Reply:
Currently we just trigger all new blocking jobs, which is technically correct but overkill -- this patch is too much in the other direction. Need to rethink